File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int finddigitProduct(int no){
        int prod=1;
        while(no!=0){
            int rem=no%10;
            prod=prod*rem;
            no/=10;
        }
        return prod;
    }
    int smallestNumber(int n, int t) {
        for(int i=n;i<=100;i++){
            int digitProduct=finddigitProduct(i);
            if(digitProduct%t==0) return i;
        }
        return -1;
    }
};

File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.py
class Solution:
    def finddigitProduct(self,no):
        prod=1
        while no!=0:
            rem=no%10
            prod=prod*rem
            no//=10
        return prod

    def smallestNumber(self, n: int, t: int) -> int:
        for i in range(n,101):
            digitProduct=self.finddigitProduct(i)
            print(digitProduct)
            if digitProduct%t==0:
                return i
        return -1
        